src/main/java/org/springframework/integration/aws/support/S3Session.java

```diff
@@ -236,7 +236,7 @@ public Object getClientInstance() {
 
 	private String[] splitPathToBucketAndKey(String path) {
 		Assert.hasText(path, "'path' must not be empty String.");
-		String[] bucketKey = path.split("/");
+		String[] bucketKey = path.split("/", 2);
```

Comment:
The same must be applied for the `list()` method as well:
```
if (bucketPrefix.length > 1) {
	listObjectsRequest.setPrefix(bucketPrefix[1]);
}
```
I might be wrong, but seems for me we can hierarchy in the bucket any difficulty.